tiledb/tests/test_multi_index-hp.py

```diff
@@ -0,0 +1,166 @@
+#
+# Property-based tests for Array.multi_index using Hypothesis
+#
+
+import tiledb
+from tiledb import SparseArray
+import numpy as np
+from numpy.testing import assert_array_equal
+
+from pathlib import Path
+import warnings
+
+import pytest
+from tiledb.tests.common import checked_path
+from tiledb.tests.common import bounded_ntuple, ranged_slices
+
+from hypothesis import given, assume, settings
+from hypothesis import strategies as st
+
+
+def is_boundserror(exc: Exception):
+    assert str(exc) != ""
+
+    vals = ["out of domain bounds", "Cannot add range to dimension"]
+
+    return any(x in str(exc) for x in vals)
+
+
+def _direct_query_ranges(array: SparseArray, ranges):
+    ctx = tiledb.default_ctx()
+    q = tiledb.core.PyQuery(ctx, array, ("a",), (), 0, False)
+    q.set_ranges(ranges)
+    q.submit()
+
+    res = {k: v[0].view(array.attr(0).dtype) for k, v in q.results().items()}
+    return res
+
+
+# Compound strategies to build valid inputs for multi_index
+tuple_indexer = st.one_of(
+    st.none(),
+    st.integers(),
+)
+subindex_obj = st.one_of(
+    st.none(),
```

Comment:
Why is this needed? Afaik we don't support indexing by None

Reply:
removed